### PR TITLE
Specify filter-data parsing

### PR DESCRIPTION
**Pull request by apasel422**
***
<a href="https://pr-preview.s3.amazonaws.com/apasel422/conversion-measurement-api/pull/468.html" title="Last updated on Jun 6, 2022, 7:05 PM UTC (85495ac)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/attribution-reporting-api/468/40bfa3a...apasel422:85495ac.html" title="Last updated on Jun 6, 2022, 7:05 PM UTC (85495ac)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -516,6 +522,31 @@ controls the maximum number of attributions for a
 
 # General Algorithms # {#general-algorithms}
 
+<h3 id="parsing-filter-data">Parsing filter data</h3>
+
+To <dfn>parse filter data</dfn> given a |value| and an optional [=boolean=]
```
Comment (csharrison) [inline index.bs:527]: Optional nit: a boolean is slightly less clear than an enum, where the value says what is for.

File: index.bs [hunk 2]
```diff
@@ -516,6 +522,31 @@ controls the maximum number of attributions for a
...
+1. If |value| is not a [=map=], return null.
+1. If |forSource| is true and |value|["`source_type`"] [=map/exists=], return
+    null.
+1. If |value|'s [=map/size=] is greater than the user agent's
+    [=max entries per attribution filter map=], return null.
+1. Let |result| be a new [=attribution filter map=].
+1. [=map/iterate|For each=] |filter| → |data| of |value|:
+    1. If |data| is not a [=list=], return null.
+    1. Let |set| be a new [=ordered set=].
+    1. [=list/iterate|For each=] |d| of |data|:
```
Comment (johnivdel) [inline index.bs:539]: Do we want to use a more specific name here like filterValueItem? I'm fine with the short form d as well though.

File: index.bs [hunk 3]
```diff
@@ -643,6 +680,7 @@ To <dfn noexport>parse source-registration JSON</dfn> given a [=string=]
         [=max attributions per event source=].
 1. Let |numReportWindows| be the result of running
     [=obtain the number of report windows=] with |sourceType|.
+1. [=map/Set=] |filterData|["`source_type`"] to « |sourceType| ».
```
Comment (csharrison) [inline index.bs:683]: Why not have this above right after invoking `parse filter data`?

File: index.bs [hunk 4]
```diff
@@ -839,6 +883,11 @@ To <dfn noexport>parse trigger-registration JSON</dfn> given a [=string=]
         <a spec="html">rules for parsing non-negative integers</a> to
         |value|["`debug_key`"].
     1. If |debugKey| is an error, set |debugKey| to null.
+1. Let |filters| be a new [=attribution filter map=].
```
Comment (johnivdel) [inline index.bs:886]: Unrelated:

Now that the headers are collapsed should we support top level negation on the trigger side? Seems like there is no reason not to.

**Review by johnivdel — APPROVED** [review]
LGTM thanks!

**Review by csharrison — APPROVED** [review]

File: index.bs (hunk 2 above)
Comment (apasel422) [reply index.bs:539]: It has a short scope, so I think it's fine as is.

File: index.bs (hunk 4 above)
Comment (apasel422) [reply index.bs:886]: We could easily do so, but let's discuss in a separate bug.

**Pull request merged**